service-impl/audit-service-impl/stroom-audit-service/src/main/java/uk.gov.gchq.palisade.audit.service.impl/StroomSimpleAuditService.java

```diff
@@ -0,0 +1,249 @@
+import uk.gov.gchq.palisade.audit.service.request.ProcessingStartedAuditRequest;
+import uk.gov.gchq.palisade.audit.service.request.ReadRequestExceptionAuditRequest;
+import uk.gov.gchq.palisade.audit.service.request.ReadRequestReceivedAuditRequest;
+import uk.gov.gchq.palisade.audit.service.request.ReadResponseAuditRequest;
+import uk.gov.gchq.palisade.audit.service.request.RequestReceivedAuditRequest;
+import uk.gov.gchq.palisade.exception.NoConfigException;
+import uk.gov.gchq.palisade.service.ServiceState;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+import static java.util.Objects.requireNonNull;
+
+
+/**
+ * A StroomAuditService is a simple implementation of an {@link AuditService} that simply constructs a message and logs
+ * it using the Stroom EventLoggingService.
+ */
+public class StroomSimpleAuditService implements AuditService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(StroomSimpleAuditService.class);
+    private static final Map<Class, Consumer<Object>> DISPATCH = new HashMap<Class, Consumer<Object>>();
+    //Create the logging service
+    private static final EventLoggingService EVENT_LOGGING_SERVICE = new DefaultEventLoggingService();
+    private static final System EVENT_LOGGING_SYSTEM = new System();
+
+
+    public static void logToStroom(final Class<? extends AuditRequest> cls, final String msg, final RequestId requestId, final UserId userId) {
+//Create the user involved in the authenticate action (possibly different from
+//the eventSource user)
+        String userMsg = "";
+        if (userId != null) {
+            userMsg += userId.toString() + " ";
+        }
+        userMsg += requestId.toString();
+
+        final User user = EventLoggingUtil.createUser(userMsg);
+        String hostName;
+        String hostAddress;
+
+        //Describe the device the event occurred on
+        try {
+            InetAddress inetAddress = InetAddress.getLocalHost();
+            hostName = inetAddress.getHostName();
+            hostAddress = inetAddress.getHostAddress();
+        } catch (UnknownHostException e) {
+            throw new RuntimeException(e);
+        }
+        final Device device = DeviceUtil.createDevice(hostName, hostAddress);
+
+
+//Provide details of where the event came from
+        final Event.EventSource eventSource = new Event.EventSource();
+        eventSource.setSystem(EVENT_LOGGING_SYSTEM);
+        eventSource.setGenerator("testCode");
+        eventSource.setDevice(device);
+        eventSource.setUser(user);
+
+//Create the authenticate object to describe the authentication specific details
+        final Event.EventDetail.Process process = new Process();
+        process.setAction(ProcessAction.EXECUTE);
+        //mmm not sure what action is the most appropriate here !!!
```

Comment:
This needs to be changed, but we shouldn't leave this in production code. This either needs to be resolved, or marked as //TODO and an issue added to the backlog.